File: Python3/0377. Combination Sum IV.py
class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        dp = [1]+[0]*(target)
        
        for i in range(target+1):
            for j in nums:
                if(i>=j):
                    dp[i] += dp[i-j]
        return dp[-1]

File: C++/0377. Combination Sum IV.cpp
class Solution {
public:
    int combinationSum4(vector<int>& nums, int target) {
        sort(nums.begin(), nums.end());
        vector<int> dp (target + 1, 0);
        dp[0] = 1;
        for(int i = 1; i <= target; ++i) {
            for(auto n : nums) {
                if(i >= n && dp[i - n] < INT_MAX - dp[i]) {
                    dp[i] += dp[i - n];
                }
            }
        }
        return dp.back();
    }
};
